ast/desugar/Desugar.cc

```diff
@@ -918,69 +929,19 @@ ExpressionPtr node2TreeImplBody(DesugarContext dctx, parser::Node *what) {
-                                                parser::isa_node<parser::Pair>(node.get()));
-                                        return parser::isa_node<parser::Kwsplat>(node.get()) ||
-                                               parser::isa_node<parser::ForwardedKwrestArg>(node.get());
-                                    })) {
-                                    // hold a reference to the node, and remove it from the back of the send list
-                                    auto node = std::move(send->args[kwargsHashIndex]);
-                                    send->args.erase(send->args.begin() + kwargsHashIndex);
-
-                                    // inline the hash into the send args
-                                    for (auto &entry : hash->pairs) {
-                                        typecase(
-                                            entry.get(),
-                                            [&](parser::Pair *pair) {
-                                                send->args.emplace_back(std::move(pair->key));
-                                                send->args.emplace_back(std::move(pair->value));
-                                            },
-                                            [&](parser::Node *node) { Exception::raise("Unhandled case"); });
-                                    }
-                                }
-                            }
-                        }
+                    if (kwargElements.has_value()) {
+                        // Concat the flattened Hash elements on the end of the args list.
+                        send->args.insert(send->args.end(), make_move_iterator(kwargElements->begin()),
+                                          make_move_iterator(kwargElements->end()));
                     }
 
                     Send::ARGS_store args;
-                    unique_ptr<parser::Node> block;
                     args.reserve(send->args.size());
-                    bool anonymousBlockPass = false;
-                    core::LocOffsets bpLoc;
                     for (auto &stat : send->args) {
-                        if (auto bp = parser::cast_node<parser::BlockPass>(stat.get())) {
-                            ENFORCE(block == nullptr, "passing a block where there is no block");
-                            if (bp->block == nullptr) {
-                                anonymousBlockPass = true;
-                                bpLoc = bp->loc;
-                            } else {
-                                block = std::move(bp->block);
-                            }
-
-                            // we don't count the block arg as part of the positional arguments anymore.
-                            numPosArgs = max(0, numPosArgs - 1);
-                        } else {
-                            args.emplace_back(node2TreeImpl(dctx, stat));
-                        }
+                        args.emplace_back(node2TreeImpl(dctx, stat));
```

Comment:
Now this just purely does the concat, where previously it was combined in with the loop that was hunting for the BlockPass.